bin/download.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+set -euo pipefail
+
+RESOURCES=$(dirname $0)/../src/main/resources/aircompressor
+
+download_linux()
+{
+    echo "Download $3 ..."
+    OUT="$RESOURCES/$3"
+    if [ -f "$OUT" ]; then
+        echo "=> skipped"
+        return
+    fi
+
+    TEMP=$(mktemp)
+    curl -sSL "$1" | tar -xO data.tar.xz | tar -xO ".$2" > "$TEMP"
+    mv -f "$TEMP" "$OUT"
+    chmod 644 "$OUT"
+    echo "=> downloaded"
+}
+
+download_macos()
+{
+    echo "Download $5 ..."
+    OUT="$RESOURCES/$5"
+    if [ -f "$OUT" ]; then
+        echo "=> skipped"
+        return
+    fi
+
+    DIGEST=$(curl -sS -L \
+        -H 'Authorization: Bearer QQ==' \
+        -H 'Accept: application/vnd.oci.image.index.v1+json' \
+        "https://ghcr.io/v2/homebrew/core/$1/manifests/$2"  \
+        | jq -r "
+            .manifests |
+            sort_by(.platform[\"os.version\"]) |
+            .[] |
+            select(.platform.os == \"darwin\") |
+            select(.platform.architecture == \"$4\") |
+            .annotations[\"sh.brew.bottle.digest\"]" \
+        | head -n1)
+
+    TEMP=$(mktemp)
+    curl -sS -L \
+        -H 'Authorization: Bearer QQ==' \
+        "https://ghcr.io/v2/homebrew/core/$1/blobs/sha256:$DIGEST" | \
+        tar -xO "$1/$2/lib/$3" > "$TEMP"
+    mv -f "$TEMP" "$OUT"
+    chmod 644 "$OUT"
+    echo "=> downloaded"
+}
+
+# Snappy
+download_linux \
+  "http://http.us.debian.org/debian/pool/main/s/snappy/libsnappy1v5_1.1.10-1+b1_amd64.deb" \
```

Comment:
Hi! I'm a nobody out of nowhere, but I'm wondering if it wouldn't be more secure to use `https` instead of `http` here?

Reply:
Unfortunately, Debian does not support https

Reply:
Maybe there's a signature to verify? Otherwise this seems like a supply chain attack waiting to happen.